### PR TITLE
Reactor + WebFlux

### PR DESCRIPTION
**Pull request by YuliaShvets**

### CODE REVIEW

File: src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingCreateController.kt [hunk 1]
```diff
@@ -22,10 +23,9 @@ class NatsParkingCreateController(
...
 
-    override fun generateReplyForNatsRequest(
-        request: ParkingRequest
-    ): ParkingResponse {
-        val parking: Parking = service.addEntity(converter.protoRequestToParking(request))
-        return converter.parkingToProtoResponse(parking)
+    override fun generateReplyForNatsRequest(request: ParkingRequest): Mono<ParkingResponse> {
+        val parking: Mono<Parking> = service.addEntity(converter.protoRequestToParking(request))
+        return parking
+            .map { converter.parkingToProtoResponse(it) }
```
Comment (ajax-shulepa-a) [inline src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingCreateController.kt]:
```suggestion
        return service.addEntity(converter.protoRequestToParking(request))
            .map { converter.parkingToProtoResponse(it) }
```

no need to split reactor chain here, try to have one chain when it's possible

File: src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingDeleteController.kt [hunk 2]
```diff
@@ -18,9 +19,8 @@ class NatsParkingDeleteController(
 
     override val parser: Parser<DeleteParkingRequest> = DeleteParkingRequest.parser()
 
-    override fun generateReplyForNatsRequest(request: DeleteParkingRequest): DeleteParkingResponse {
-        val parkingId = request.parkingId
-        service.deleteEntity(parkingId)
-        return DeleteParkingResponse.newBuilder().build()
+    override fun generateReplyForNatsRequest(request: DeleteParkingRequest): Mono<DeleteParkingResponse> {
+        service.deleteEntity(request.parkingId).block()
+        return Mono.just(DeleteParkingResponse.getDefaultInstance())
```
Comment (ajax-shulepa-a) [inline src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingDeleteController.kt]:
```suggestion
        return service.deleteEntity(request.parkingId)
            .thenReturn(DeleteParkingResponse.getDefaultInstance())
```
do not use block in your code, you're multiplying by zero whole async-non-blocking concepts. The only place where it is possible is the test code.

File: src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingGetByIdController.kt [hunk 3]
```diff
@@ -20,15 +21,15 @@ class NatsParkingGetByIdController(
...
-            .setParking(protoParking)
-            .build()
+        return service.findEntityById(parkingId)
+            .map { parking ->
+                val protoParking = converter.parkingToProto(parking)
+                GetByIdParkingResponse.newBuilder()
+                    .setParking(protoParking)
+                    .build()
+            }
     }
```
Comment (ajax-shulepa-a) [inline src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingGetByIdController.kt]:
```suggestion
        return service.findEntityById(parkingId)
            .map { converter.parkingToProto(it) }
            .map {
                GetByIdParkingResponse.newBuilder()
                    .setParking(it)
                    .build()
            }
    }
```

you have two mapping operations and it's better to split them into two reactor steps. also, do not afraid to use it

File: src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingUpdateController.kt [hunk 4]
```diff
@@ -20,20 +21,22 @@ class NatsParkingUpdateController(
...
-            )
-            .build()
+
+        return Mono.fromCallable {
+            val createdParking = service.updateEntity(parking)
+
+            ParkingOuterClass.UpdateParkingResponse.newBuilder()
+                .setParking(
+                    converter
+                        .parkingToProto(createdParking.block()!!) // block to get the result from Mono
```
Comment (ajax-shulepa-a) [inline src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingUpdateController.kt]: do not block, refactor it please

File: src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingUpdateController.kt [hunk 5]
```diff
@@ -20,20 +21,22 @@ class NatsParkingUpdateController(
...
-
-        return ParkingOuterClass.UpdateParkingResponse.newBuilder()
-            .setParking(
-                converter
-                    .parkingToProto(createdParking)
-            )
-            .build()
+
+        return Mono.fromCallable {
+            val createdParking = service.updateEntity(parking)
```
Comment (ajax-shulepa-a) [inline src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingUpdateController.kt]: it's better not to execute some logic outside the reactor chain, then when something fails, an error will be propagated in the reactor chain. so move your protoToParking mapping to reactor chain

**Review by ajax-shulepa-a — CHANGES_REQUESTED** [review]

File: src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingCreateController.kt (hunk 1 above)
Comment (YuliaShvets) [reply src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingCreateController.kt]: ✅

File: src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingGetByIdController.kt (hunk 3 above)
Comment (YuliaShvets) [reply src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingGetByIdController.kt]: ✅

File: src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingUpdateController.kt (hunk 5 above)
Comment (YuliaShvets) [reply src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingUpdateController.kt]: ✅

File: src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingUpdateController.kt (hunk 4 above)
Comment (YuliaShvets) [reply src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingUpdateController.kt]: ✅

File: src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingDeleteController.kt (hunk 2 above)
Comment (YuliaShvets) [reply src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingDeleteController.kt]: ✅

**Review by ajax-shulepa-a — APPROVED** [review]

**Pull request merged**